src/SilentRenewService.ts

```diff
@@ -44,7 +49,15 @@ export class SilentRenewService {
         try {
             await this._userManager.signinSilent();
             logger.debug("silent token renewal successful");
-        } catch (err) {
+        }
+        catch (err) {
+            if (this._userManager.settings.retrySilentRenew && !(err instanceof ErrorResponse)) {
```

Comment:
I think it's not OK to always retry - some errors must stop the silent retrying - for example if consent is required or if not signed in at all.

Reply:
If the authorization server is behaving correctly, this won't retry since the error would an `ErrorResponse` instance with `err.error === 'consent_required'` or `err.error === 'login_required'`, etc.